### PR TITLE
application: serial_lte_modem: SLM restructuring

### PR DESCRIPTION
**Pull request by junqingzou**
Restruct project as below:
Default services - Generic, TCP/UDP proxy, Socket, ICMP Ping, FOTA
Configurable services - GPS, FTP client, MQTT client, HTTP client
Configurable solutions - native TLS support, SUPL-based A-GPS

JIRA: NCSIDB-240

Signed-off-by: Jun Qing Zou <jun.qing.zou@nordicsemi.no>

### CODE REVIEW

File: applications/serial_lte_modem/sample.yaml [hunk 1]
```diff
@@ -6,19 +6,19 @@ tests:
     build_on_all: true
     platform_allow: nrf9160dk_nrf9160ns
     tags: ci_build
-  samples.nrf9160.serial_lte_modem.tcp_udp_server:
+  samples.nrf9160.serial_lte_modem.full_services:
```
[the remark this answers is not in the document]
Comment (junqingzou) [reply applications/serial_lte_modem/sample.yaml:9]: They are not enabled in prj.conf by default

[the remark this answers is not in the document]
Comment (rlubos) [reply applications/serial_lte_modem/sample.yaml:9]: Ah right, now I see they're commented out. Sorry for the fuss.

**Review by rlubos — APPROVED** [review]

**Review by lats1980 — APPROVED** [review]
Looks good to me.

**Review by ru-fu — CHANGES_REQUESTED** [review]
Sorry for the delay - I'll add a commit with doc changes.

**Review by ru-fu — COMMENTED** [review]
I added my commit - please check and squash if okay. :)

**Comment by junqingzou** [comment]
> I added my commit - please check and squash if okay. :)

All changes are included in the new commit. Thanks a lot!

File: applications/serial_lte_modem/doc/slm_description.rst [hunk 2]
```diff
@@ -1645,6 +1662,7 @@ Complete the following steps to test the functionality provided by the :ref:`SLM
 Testing FTP AT commands
 -----------------------
 
+Note that these commands are available only if :option:`CONFIG_SLM_FTPC` is defined.
```
Comment (ru-fu) [inline applications/serial_lte_modem/doc/slm_description.rst:1665]: A note like this must also be included in the (new) GPS section.

Comment (junqingzou) [reply applications/serial_lte_modem/doc/slm_description.rst:1665]: Added

Comment (ru-fu) [reply applications/serial_lte_modem/doc/slm_description.rst:1665]: I can't see this change, did you maybe not push it?

Comment (junqingzou) [reply applications/serial_lte_modem/doc/slm_description.rst:1665]: It's a shame that I committed without including the change.
Now fixed

**Review by ru-fu — APPROVED** [review]

**Pull request merged**